File: Practica1/Ejercicio2/FactoriaCarretera.py
from FactoriaCarreraBicicleta import FactoriaCarreraBicicleta
from CarreraCarretera import CarreraCarretera
from BicicletaCarretera import BicicletaCarretera

class FactoriaCarretera(FactoriaCarreraBicicleta):

    def crear_carrera(self, bicicletas, porcentaje):
        return CarreraCarretera(bicicletas, porcentaje)

    def crear_bicicleta(self, id):
        return BicicletaCarretera(id)

File: Practica1/Ejercicio1/FactoriaCarretera.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package java;

/**
 *
 * @author Usuario
 */
import prac1.FactoriaCarreraBicicleta;
import prac1.bicicletaMontana;
import prac1.bicicletaCarretera;
import prac1.carreraCarretera;
import java.util.ArrayList;
public class FactoriaCarretera implements FactoriaCarreraBicicleta{
          
    @Override
    public bicicleta crearBicicleta(int i){
            return new bicicletaCarretera(i);       
    }
    
    @Override
    public carrera crearCarrera(ArrayList<bicicleta> participantes){
        return new carreraCarretera(participantes);
        
    }
        
}
